packages/next-server/server/config.ts

```diff
@@ -17,6 +17,8 @@ const defaultConfig: { [key: string]: any } = {
   pageExtensions: ['tsx', 'ts', 'jsx', 'js'],
   target: process.env.__NEXT_BUILDER_EXPERIMENTAL_TARGET || 'server',
   poweredByHeader: true,
+  modern: true,
+  safari10NomoduleFix: false,
```

Comment:
Should this option be true by default, so that Next.js just works™️? Opting out makes sense when devs are sure that the app is known to be executed only in certain browsers. This is not the case for publicly facing websites. To make the option 'truly' when set, it can be called `skipSafari10NomoduleFix`.

Reply:
Hey @kachkaev. The reason I kept this to false by default was that this is an issue for a [very small percentage of users](https://browserl.ist/?q=Safari+10.1)
Apart from the the size of the script itself, it attaches an event handler for all script loads. Although insignificant I thought it's best the devs opt in for the feature.

Reply:
I agree that the global proportion of Safari 10.1 users is rather small. However, it may vary significantly for different projects. E.g. if some website is targeting vintage mac holders, it's easy to imagine that this proportion will be way-way more.

I have a project where the proportion of IE11 users is crazy, but that's because its domain targets a certain social and economic group. People are on old Windows machines, which explains it.

If someone is developing a website and is new to Next.js, how do they start? They look at the list of supported browsers, see that it's IE11 or greater, get happy and proceed to `next dev`. If Safari module fixing is off by default, the browser range in README would have to contain a disclaimer, explaining some odd caveat that applies to some specific Safari version. The developer will be like: _Is Safari 10.1 the latest? Or is it not? I'd rather check. Oh, it's not so much prevalent, but who knows maybe my users will come from this browser often (my website will be BIG). I'd probably rather turn that thing on. How do I do this? Oh, here is this other section of the docs about the config, hmmm. How does this even work? StackOverflow, HALP!_ At least 30 minutes are lost even before the first page was written. And this scenario will be quite likely for someone starting their first Next.js project.

Alternative scenario: A team ships a website, it becomes successful, but then they start getting some odd user feedback from a small proportion of customers and go investigating. Half a day is lost + the website did not 'sell' stuff for hundreds of people alread.

Now what will happen if Safari fix is on by default: Bundle size is increased by ≈100-200 bytes for everyone, plus there exists a small script event listener. Both seem nothing in the scale of the entire app. When can I as a developer 'lose time'? Perhaps, only if I go to the app's generated HTML page source, read it carefully, notice some odd code block and become curious about what it does. I search for it, come across the option and turn it off to save a few bytes of my app size. What proportion of developers will go down this road? My guess that this number will be comparable with global Safari 10.1 users 😀

Reply:
Thanks for the feedback. That makes sense. I've changed it to load by default.

I've also added a noModule attribute to the script, which largely solves the concerns I had!